crates/nu-command/src/math/utils.rs

```diff
@@ -80,15 +80,24 @@ pub fn calculate(
             ),
             _ => mf(vals, span, name),
         },
-        PipelineData::Value(Value::Record { val: record, .. }, ..) => {
-            let mut record = record;
+        PipelineData::Value(mut val @ Value::Record { .. }, ..) => {
+            let Value::Record {
+                val: ref mut record,
+                ..
+            } = val
+            else {
+                // We already verified that it's a record, but can't pattern match inside at the
+                // same time.
+                unreachable!()
+            };
             record
+                .to_mut()
                 .iter_mut()
                 .try_for_each(|(_, val)| -> Result<(), ShellError> {
                     *val = mf(slice::from_ref(val), span, name)?;
                     Ok(())
                 })?;
-            Ok(Value::record(*record, span))
+            Ok(val)
```

Comment:
I think the previous is more clear (and has no `unreachable!` usage).

Reply:
Yeah, you're right, I don't know what I was thinking here. Taking with `.into_owned()` is just as good.